src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -394,13 +394,29 @@ void MergeLibraryManifest (string mergedManifest)
 					doc.XPathSelectElement (string.Format ("/manifest/{0}", top.Name.LocalName));
 				if (existing != null)
 					// if there is existing node with the same android:name, then append contents to existing node.
-					existing.Add (FixupNameElements (package, top.Nodes ()));
+					MergeNodesIntoElement (existing, FixupNameElements (package, top.Nodes ()));
 				else
 					// otherwise, just add to the doc.
 					doc.Root.Add (FixupNameElements (package, new XNode [] { top }));
 			}
 		}
 
+		bool AttributesMatch (XElement a, XElement b)
+		{
+			return a.Attributes ().All (x => b.Attribute (x.Name.LocalName).Value == x.Value); ;
+		}
+
+		void MergeNodesIntoElement (XElement element, IEnumerable<XNode> nodes)
+		{
+			var name = AndroidXmlNamespace.GetName ("name");
+			foreach (var node in nodes.Select (x => x as XElement)) {
+				var existing = element.XPathSelectElements ($"./{node.Name.LocalName}").Any (x => AttributesMatch (x, node));
```

Comment:
I don't see why this needs to use `XPathSelectElements()` when it's for a single element. Shouldn't `element.Elements (node.Name)` work?